{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MLPA Data Model\n",
    "\n",
    "MLPA reports have 4 data types:\n",
    "\n",
    "* Tests - varchar(200)\n",
    "    * Methods (text)\n",
    "* Genes - varchar(200)\n",
    "* Variations - varchar(200)\n",
    "* Results - text\n",
    "* Interpretations - text\n",
    "\n",
    "A test has a methods\n",
    "\n",
    "A test has many genes\n",
    "\n",
    "A test has many variations\n",
    "\n",
    "A variation has many results\n",
    "\n",
    "A result has many interpretations\n",
    "\n",
    "```\n",
    "GeneID : {\n",
    "    variation1 : {\n",
    "                 result : interpretation\n",
    "               }\n",
    "   variation1: {\n",
    "                 result : interpretation\n",
    "               }\n",
    "}  \n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example MLPA Test\n",
    "\n",
    "**Test Name** \n",
    "\n",
    "Alpha Thalassemia: HBA1&HBA2 Deletion/Duplication Analysis\n",
    "\n",
    "**Method**\n",
    "\n",
    "DNA analysis of the alpha-globin locus (HBA1/HBA2, OMIM 141800/141850, 16pter - 16p13.3)  is performed by multiplex ligation-dependent probe amplification (MLPA). This methodology will, in principle, detect all genomic deletions and duplications involving this locus, as well as the Constant Spring point mutation. Any partial exonic deletions and duplications outside of the region of interest cannot be detected. The analysis does not detect other point mutations within alpha globin locus. The presence of a rare mutation cannot be entirely ruled out. Any interpretation given here should be clinically correlated with available information about presentation and relevant family history of the patient.\n",
    "\n",
    "**GENE IDS**\n",
    "\n",
    "HBA1 and HBA2\n",
    "\n",
    "**Variation**\n",
    "\n",
    "\"Heterozygous α3.7 deletion\" \n",
    "\n",
    "**Result:** \n",
    "\n",
    "\"Heterozygous α3.7 deletion: One copy of the α3.7 deletion was detected in the analyzed region.\"  \n",
    "\n",
    "\n",
    "**Interpretation** \n",
    "\n",
    "\"This result is most consistent with this patient being an unaffected carrier of alpha-thalassemia with a single gene deletion, also called alpha + thalassemia trait. Individuals with alpha + thalassemia trait typically have no clinical findings. Genetic counselor is available to health care providers  to discuss this results further. \""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example Test\n",
    "\n",
    "**Name**\n",
    "\n",
    "Alpha Thalassemia: HBA1&HBA2 Deletion/Duplication Analysis\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_df_records = [\n",
    "    {\n",
    "        \"id\": 1,\n",
    "        \"name\": \"Alpha Thalassemia: HBA1&HBA2 Deletion/Duplication Analysis\",\n",
    "        \"method\": \"\"\"DNA analysis of the alpha-globin locus (HBA1/HBA2, OMIM 141800/141850, 16pter - 16p13.3)  is performed by multiplex ligation-dependent probe amplification (MLPA). This methodology will, in principle, detect all genomic deletions and duplications involving this locus, as well as the Constant Spring point mutation. Any partial exonic deletions and duplications outside of the region of interest cannot be detected. The analysis does not detect other point mutations within alpha globin locus. The presence of a rare mutation cannot be entirely ruled out. Any interpretation given here should be clinically correlated with available information about presentation and relevant family history of the patient.\"\"\"\n",
    "    }\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_df = pd.DataFrame.from_records(test_df_records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>method</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Alpha Thalassemia: HBA1&amp;HBA2 Deletion/Duplicat...</td>\n",
       "      <td>DNA analysis of the alpha-globin locus (HBA1/H...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                name  \\\n",
       "0  Alpha Thalassemia: HBA1&HBA2 Deletion/Duplicat...   \n",
       "\n",
       "                                              method  \n",
       "0  DNA analysis of the alpha-globin locus (HBA1/H...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Example Gene\n",
    "\n",
    "Genes - HBA1&HBA2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>HBA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>HBA2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id  name\n",
       "0   1  HBA1\n",
       "1   2  HBA2"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_records = [\n",
    "    {\n",
    "        \"id\": 1,\n",
    "        \"name\": \"HBA1\"\n",
    "    },\n",
    "    {\n",
    "        \"id\": 2,\n",
    "        \"name\": \"HBA2\"\n",
    "    }\n",
    "]\n",
    "gene_df = pd.DataFrame.from_records(gene_records)\n",
    "gene_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>test_id</th>\n",
       "      <th>variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>α3.7 deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous α4.2 deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous HbCS variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous α3.7 deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous α4.2 deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous HbCS variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous α3.7 deletion &amp; Heterozygous α4.2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous α3.7 deletion &amp; Heterozygous HbCS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous α4.2 deletion &amp; Heterozygous HbCS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous --SEA deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous --MED deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous --FIL deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous --THAI deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous α3.7 deletion &amp; Heterozygous --SE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous α4.2 deletion &amp; Heterozygous --SE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous HbCS variant &amp; Heterozygous --SEA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous α3.7 deletion &amp; Heterozygous --ME...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous α4.2 deletion &amp; Heterozygous --ME...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>1</td>\n",
       "      <td>Heterozygous HbCS variant &amp; Heterozygous --MED...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous --SEA deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous --MED deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous --FIL deletion</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>1</td>\n",
       "      <td>Homozygous --THAI deletion</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    id  test_id                                          variation\n",
       "0    1        1                                      α3.7 deletion\n",
       "1    2        1                         Heterozygous α4.2 deletion\n",
       "2    3        1                          Heterozygous HbCS variant\n",
       "3    4        1                           Homozygous α3.7 deletion\n",
       "4    5        1                           Homozygous α4.2 deletion\n",
       "5    6        1                            Homozygous HbCS variant\n",
       "6    7        1  Heterozygous α3.7 deletion & Heterozygous α4.2...\n",
       "7    8        1  Heterozygous α3.7 deletion & Heterozygous HbCS...\n",
       "8    9        1  Heterozygous α4.2 deletion & Heterozygous HbCS...\n",
       "9   10        1                        Heterozygous --SEA deletion\n",
       "10  11        1                        Heterozygous --MED deletion\n",
       "11  12        1                        Heterozygous --FIL deletion\n",
       "12  13        1                       Heterozygous --THAI deletion\n",
       "13  14        1  Heterozygous α3.7 deletion & Heterozygous --SE...\n",
       "14  15        1  Heterozygous α4.2 deletion & Heterozygous --SE...\n",
       "15  16        1  Heterozygous HbCS variant & Heterozygous --SEA...\n",
       "16  17        1  Heterozygous α3.7 deletion & Heterozygous --ME...\n",
       "17  18        1  Heterozygous α4.2 deletion & Heterozygous --ME...\n",
       "18  19        1  Heterozygous HbCS variant & Heterozygous --MED...\n",
       "19  20        1                          Homozygous --SEA deletion\n",
       "20  21        1                          Homozygous --MED deletion\n",
       "21  22        1                          Homozygous --FIL deletion\n",
       "22  23        1                         Homozygous --THAI deletion"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variations = \"\"\"α3.7 deletion\n",
    "Heterozygous α4.2 deletion\n",
    "Heterozygous HbCS variant\n",
    "Homozygous α3.7 deletion\n",
    "Homozygous α4.2 deletion\n",
    "Homozygous HbCS variant\n",
    "Heterozygous α3.7 deletion & Heterozygous α4.2 deletion\n",
    "Heterozygous α3.7 deletion & Heterozygous HbCS variant\n",
    "Heterozygous α4.2 deletion & Heterozygous HbCS variant\n",
    "Heterozygous --SEA deletion\n",
    "Heterozygous --MED deletion\n",
    "Heterozygous --FIL deletion\n",
    "Heterozygous --THAI deletion\n",
    "Heterozygous α3.7 deletion & Heterozygous --SEA deletion\n",
    "Heterozygous α4.2 deletion & Heterozygous --SEA deletion\n",
    "Heterozygous HbCS variant & Heterozygous --SEA deletion\n",
    "Heterozygous α3.7 deletion & Heterozygous --MED deletion\n",
    "Heterozygous α4.2 deletion & Heterozygous --MED deletion\n",
    "Heterozygous HbCS variant & Heterozygous --MED deletion\n",
    "Homozygous --SEA deletion\n",
    "Homozygous --MED deletion\n",
    "Homozygous --FIL deletion\n",
    "Homozygous --THAI deletion\"\"\"\n",
    "\n",
    "variations = variations.split(\"\\n\")\n",
    "variation_records = []\n",
    "x = 1\n",
    "for variation in variations:\n",
    "    variation_records.append({\n",
    "      \"id\": x,\n",
    "      \"test_id\": 1  ,\n",
    "      \"variation\": variation,\n",
    "    })\n",
    "    x = x+1\n",
    "\n",
    "variations_df = pd.DataFrame.from_records(variation_records)\n",
    "variations_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_gene_records = [\n",
    "    {\"gene_id\": 1, \"test_id\": 1},\n",
    "    {\"gene_id\": 2, \"test_id\": 1},\n",
    "]\n",
    "\n",
    "test_gene_df = pd.DataFrame.from_records(test_gene_records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>test_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>result</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>Negative: No deletions were detected within th...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>Heterozygous α3.7 deletion: One copy of the α3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>Heterozygous α4.2 deletion: One copy of the α4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>Heterozygous HbCS variant: One copy of the Hb ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>Homozygous α3.7 deletion: Two copies of the α3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>Homozygous α4.2 deletion: Two copies of the α4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>Homozygous HbCS variant: Two copies of the Hb ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>Heterozygous α3.7 deletion &amp; Heterozygous α4.2...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>Heterozygous α3.7 deletion &amp; Heterozygous HbCS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>Heterozygous α4.2 deletion &amp; Heterozygous HbCS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>Heterozygous --SEA deletion: One copy of the -...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>Heterozygous --MED deletion: One copy of the -...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>Heterozygous --FIL deletion: One copy of the -...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>Heterozygous --THAI deletion: One copy of the ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1</td>\n",
       "      <td>15</td>\n",
       "      <td>Heterozygous α3.7 deletion &amp; Heterozygous --SE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>Heterozygous α4.2 deletion &amp; Heterozygous --SE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1</td>\n",
       "      <td>17</td>\n",
       "      <td>Heterozygous HbCS variant &amp; Heterozygous --SEA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1</td>\n",
       "      <td>18</td>\n",
       "      <td>Heterozygous α3.7 deletion &amp; Heterozygous --ME...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>Heterozygous α4.2 deletion &amp; Heterozygous --ME...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>Heterozygous HbCS variant &amp; Heterozygous --MED...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1</td>\n",
       "      <td>21</td>\n",
       "      <td>Homozygous --SEA deletion:Two copies of the --...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1</td>\n",
       "      <td>22</td>\n",
       "      <td>Homozygous --MED deletion: Two copies of the -...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>1</td>\n",
       "      <td>23</td>\n",
       "      <td>Homozygous --FIL deletion: Two copies of the -...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>1</td>\n",
       "      <td>24</td>\n",
       "      <td>Homozygous --THAI deletion: Two copies of the ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    test_id  variation_id                                             result\n",
       "0         1             1  Negative: No deletions were detected within th...\n",
       "1         1             2  Heterozygous α3.7 deletion: One copy of the α3...\n",
       "2         1             3  Heterozygous α4.2 deletion: One copy of the α4...\n",
       "3         1             4  Heterozygous HbCS variant: One copy of the Hb ...\n",
       "4         1             5  Homozygous α3.7 deletion: Two copies of the α3...\n",
       "5         1             6  Homozygous α4.2 deletion: Two copies of the α4...\n",
       "6         1             7  Homozygous HbCS variant: Two copies of the Hb ...\n",
       "7         1             8  Heterozygous α3.7 deletion & Heterozygous α4.2...\n",
       "8         1             9  Heterozygous α3.7 deletion & Heterozygous HbCS...\n",
       "9         1            10  Heterozygous α4.2 deletion & Heterozygous HbCS...\n",
       "10        1            11  Heterozygous --SEA deletion: One copy of the -...\n",
       "11        1            12  Heterozygous --MED deletion: One copy of the -...\n",
       "12        1            13  Heterozygous --FIL deletion: One copy of the -...\n",
       "13        1            14  Heterozygous --THAI deletion: One copy of the ...\n",
       "14        1            15  Heterozygous α3.7 deletion & Heterozygous --SE...\n",
       "15        1            16  Heterozygous α4.2 deletion & Heterozygous --SE...\n",
       "16        1            17  Heterozygous HbCS variant & Heterozygous --SEA...\n",
       "17        1            18  Heterozygous α3.7 deletion & Heterozygous --ME...\n",
       "18        1            19  Heterozygous α4.2 deletion & Heterozygous --ME...\n",
       "19        1            20  Heterozygous HbCS variant & Heterozygous --MED...\n",
       "20        1            21  Homozygous --SEA deletion:Two copies of the --...\n",
       "21        1            22  Homozygous --MED deletion: Two copies of the -...\n",
       "22        1            23  Homozygous --FIL deletion: Two copies of the -...\n",
       "23        1            24  Homozygous --THAI deletion: Two copies of the ..."
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results = \"\"\"Negative: No deletions were detected within the alpha globin cluster region.\n",
    "Heterozygous α3.7 deletion: One copy of the α3.7 deletion was detected in the analyzed region.\n",
    "Heterozygous α4.2 deletion: One copy of the α4.2 deletion was detected in the analyzed region.\n",
    "Heterozygous HbCS variant: One copy of the Hb constant spring (HbCS) variant (heterozygous) was detected in the analyzed region.\n",
    "Homozygous α3.7 deletion: Two copies of the α3.7 deletion were detected in the analyzed region.\n",
    "Homozygous α4.2 deletion: Two copies of the α4.2 deletion were detected in the analyzed region.\n",
    "Homozygous HbCS variant: Two copies of the Hb constant spring (HbCS) variant (homozygous) were detected in the analyzed region.\n",
    "Heterozygous α3.7 deletion & Heterozygous α4.2 deletion: One copy of the α3.7 deletion and one copy of the α4.2 deletion were detected in the analyzed region.\n",
    "Heterozygous α3.7 deletion & Heterozygous HbCS variant : One copy of the α3.7 deletion and one copy of the Hb constant spring (HbCS) variant were detected in the analyzed region.\n",
    "Heterozygous α4.2 deletion & Heterozygous HbCS variant : One copy of the α4.2 deletion and one copy of the Hb constant spring (HbCS) variant were detected in the analyzed region.\n",
    "Heterozygous --SEA deletion: One copy of the --SEA deletion was detected in the analyzed region.\n",
    "Heterozygous --MED deletion: One copy of the --MED deletion was detected in the analyzed region.\n",
    "Heterozygous --FIL deletion: One copy of the --FIL deletion was detected in the analyzed region.\n",
    "Heterozygous --THAI deletion: One copy of the --THAI deletion was detected in the analyzed region.\n",
    "Heterozygous α3.7 deletion & Heterozygous --SEA deletion: One copy of the α3.7 deletion and one copy of the --SEA deletion were detected in the analyzed region.\n",
    "Heterozygous α4.2 deletion & Heterozygous --SEA deletion: One copy of the α4.2 deletion and one copy of the --SEA deletion were detected in the analyzed region.\n",
    "Heterozygous HbCS variant & Heterozygous --SEA deletion: One copy of the HbCS variant and one copy of the --SEA deletion were detected in the analyzed region.\n",
    "Heterozygous α3.7 deletion & Heterozygous --MED deletion: One copy of the α3.7 deletion and one copy of the --MED deletion were detected in the analyzed region.\n",
    "Heterozygous α4.2 deletion & Heterozygous --MED deletion: One copy of the α4.2 deletion and one copy of the --MED deletion were detected in the analyzed region.\n",
    "Heterozygous HbCS variant & Heterozygous --MED deletion: One copy of the HbCS variant and one copy of the --MED deletion were detected in the analyzed region.\n",
    "Homozygous --SEA deletion:Two copies of the --SEA deletion was detected in the analyzed region.\n",
    "Homozygous --MED deletion: Two copies of the --MED deletion was detected in the analyzed region.\n",
    "Homozygous --FIL deletion: Two copies of the --FIL deletion was detected in the analyzed region.\n",
    "Homozygous --THAI deletion: Two copies of the --THAI deletion was detected in the analyzed region.\"\"\"\n",
    "\n",
    "results = results.split(\"\\n\")\n",
    "results_records = []\n",
    "\n",
    "x = 1\n",
    "for result in results:\n",
    "    results_records.append({\n",
    "        \"test_id\": 1,\n",
    "        \"variation_id\": x,\n",
    "        \"result\": result,\n",
    "    })\n",
    "    x = x+1\n",
    "    \n",
    "results_df = pd.DataFrame.from_records(results_records)\n",
    "results_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>result_id</th>\n",
       "      <th>interpretation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>This is a dummy interpretation that should be ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   id  result_id                                     interpretation\n",
       "0   1          1  This is a dummy interpretation that should be ..."
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "interpretation = \"\"\"This is a dummy interpretation that should be created by a real genetic counselor. Genetic counselor is available to health care providers to discuss this results further.\"\"\"\n",
    "\n",
    "interpretation_records = [{\n",
    "    \"id\": 1,\n",
    "    # I'm making up the result id\n",
    "    \"result_id\": 1,\n",
    "    \"interpretation\": interpretation,\n",
    "}]\n",
    "\n",
    "interpretation_df = pd.DataFrame.from_records(interpretation_records)\n",
    "interpretation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
